### DISCUSSION TITLE
Stream offset without offset tracking

### DISCUSSION THREAD

**Discussion by laststem** [rabbitmq/rabbitmq-server#11850] [Questions] [answered]
In [docs](https://www.rabbitmq.com/docs/streams#limitations-ui-metrics), Offset tracking information may affect offset.

In [docs](https://rabbitmq.github.io/rabbitmq-stream-java-client/stable/htmlsingle/#considerations-on-offset-tracking), Message offsets may not be contiguous.

However, If I don't use offset tracking, will the offset increase by 1 continuously?

**Comment by acogoluegnes** [comment, marked as answer]
[Message deduplication](https://rabbitmq.github.io/rabbitmq-stream-java-client/stable/htmlsingle/#outbound-message-deduplication) also affects offset contiguity.

Message offsets should be contiguous if neither message deduplication nor offset tracking is used (I don't see other cases off the top of my head). It is unlikely to change soon but we cannot guarantee it will always be like this.
